### PR TITLE
Fix compilation with Xcode 15

### PR DESCRIPTION
**Pull request by tgoyne**

### CODE REVIEW

File: RealmSwift/Map.swift [hunk 1]
```diff
@@ -814,7 +814,7 @@ extension Map: Sequence {
 
 extension Map {
     /// An adaptor for Map which makes it a sequence of `(key: Key, value: Value)` instead of a sequence of `SingleMapEntry`.
-    public struct KeyValueSequence<Key: _MapKey, Value: RealmCollectionValue>: Sequence {
```
Comment (tgoyne) [inline RealmSwift/Map.swift:817]: Duplicating the generic arguments here was incorrect and now produces a warning.

File: RealmSwift/Util.swift [hunk 2]
```diff
@@ -160,14 +160,16 @@ internal func logRuntimeIssue(_ message: StaticString) {
     }
 }
 
-#if swift(<5.9)
```
Comment (tgoyne) [inline RealmSwift/Util.swift:163]: The API ended up getting changed after I wrote this, but it turns out we can't unconditionally use the built-in versions anyway due to the deployment target.

**Review by adbdehghan — APPROVED** [review]
Please approve this pr

**Comment by davidkrammer** [comment]
Please approve this

**Review by dianaafanador3 — APPROVED** [review]

**Comment by DaniilDavidovich** [comment]
this is a working solution

**Pull request merged**